File: utils/categories.ts
import { CategoryName } from "types/categories"

export const idToCategoryMapping: {
  [key: number]: CategoryName
} = {
  1: "FAMILY",
  2: "PETS",
  3: "SOCIAL_INTERESTS",
  4: "EDUCATION_CAREER",
  5: "TRAVEL",
  6: "HEALTH_BEAUTY",
  7: "HOME_GARDEN",
  8: "FINANCE",
  9: "TRANSPORT"
}

export const categoryToIdMapping: {
  [key in CategoryName]: number
} = {
  "FAMILY": 1,
  "PETS": 2,
  "SOCIAL_INTERESTS": 3,
  "EDUCATION_CAREER": 4,
  "TRAVEL": 5,
  "HEALTH_BEAUTY": 6,
  "HOME_GARDEN": 7,
  "FINANCE": 8,
  "TRANSPORT": 9
}

File: vuetapi/core/utils/categories.py
"""Categories"""

from enum import Enum


class Categories(Enum):
    """Categories"""

    PETS = 1
    SOCIAL_INTERESTS = 2
    EDUCATION = 3
    CAREER = 4
    TRAVEL = 5
    HEALTH_BEAUTY = 6
    HOME = 7
    GARDEN = 8
    FOOD = 9
    LAUNDRY = 10
    FINANCE = 11
    TRANSPORT = 12


CATEGORY_READABLE_NAMES = {
    Categories.PETS: "Pets",
    Categories.SOCIAL_INTERESTS: "Social",
    Categories.EDUCATION: "Education",
    Categories.CAREER: "Career",
    Categories.TRAVEL: "Travel",
    Categories.HEALTH_BEAUTY: "Health & Beauty",
    Categories.HOME: "Home",
    Categories.GARDEN: "Garden",
    Categories.FOOD: "Food",
    Categories.LAUNDRY: "Laundry",
    Categories.FINANCE: "Finance",
    Categories.TRANSPORT: "Transport",
}


class CategorySettings:
    """CategorySettings"""

    def __init__(self, id, is_enabled=True, is_premium=False):
        self.id = id
        self.name = Categories(id).name
        self.readable_name = CATEGORY_READABLE_NAMES[Categories(id)]
        self.is_enabled = is_enabled
        self.is_premium = is_premium


CATEGORY_SETTINGS = {
    Categories.HEALTH_BEAUTY: CategorySettings(6, is_premium=True),
}


def get_category_settings(category_id):
    """get_category_settings"""
    category_id_int = int(category_id)
    if Categories(category_id_int) in CATEGORY_SETTINGS:
        return CATEGORY_SETTINGS.get(Categories(category_id_int))
    return CategorySettings(category_id_int)
